Error: The terraform-provider-cloudflare_v3.14.0 plugin crashed!

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform 1.1.9
Cloudflare provider 3.14.0

### Affected resource(s)

cloudflare_spectrum_application

### Terraform configuration files

```hcl
resource "cloudflare_spectrum_application" "endpoint" {
  for_each = {
    for endpoint in local.edge_endpoints : endpoint.name => endpoint
  }
  zone_id      = cloudflare_zone.main.id
  traffic_type = "direct"

  dns {
    type = each.value.edge.ips != null ? "ADDRESS" : "CNAME"
    name = each.value.edge.dns_name
  }
  edge_ip_connectivity = "all"
  edge_ips             = each.value.edge.ips
  protocol             = "tcp/${each.value.edge.port}"

  origin_dns {
    name = cloudflare_load_balancer.main.name
  }
  origin_port    = each.value.origin.port
  tls            = "off"
  ip_firewall    = false
  proxy_protocol = var.proxy_protocol ? "v2" : "off"
}
```


### Debug output

2022-05-10T16:32:50.492Z [INFO]  provider.terraform-provider-cloudflare_v3.11.0: 2022/05/10 16:32:50 [DEBUG] Cloudflare API Request Details:
---[ REQUEST ]---------------------------------------
GET /client/v4/zones/017e0312e7577130f17ca52d5bf08749/spectrum/apps/a6787f81343f4f1792d77c39bd972401 HTTP/1.1
Host: api.cloudflare.com
User-Agent: terraform/1.1.9 terraform-plugin-sdk/2.10.1 terraform-provider-cloudflare/3.11.0
Content-Type: application/json
X-Auth-Email: xxx
X-Auth-Key: xxx
Accept-Encoding: gzip


-----------------------------------------------------: timestamp=2022-05-10T16:32:50.492Z
2022-05-10T16:32:50.707Z [INFO]  provider.terraform-provider-cloudflare_v3.11.0: 2022/05/10 16:32:50 [DEBUG] Cloudflare API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Cf-Cache-Status: DYNAMIC
Cf-Ray: 709412bbcd7526db-MSP
Content-Type: application/json; charset=UTF-8
Date: Tue, 10 May 2022 16:32:50 GMT
Expect-Ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
Server: cloudflare
Set-Cookie: __cflb=0H28vgHxwvgAQtjUGU4vq74ZFe3sNVUZa4ACfsDBtGd; SameSite=Lax; path=/; expires=Tue, 10-May-22 19:02:51 GMT; HttpOnly
Set-Cookie: __cfruid=3caaba9eb5cace4011b08c2b1b72e9cab14484ca-1652200370; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
Vary: Accept-Encoding
X-Envoy-Upstream-Service-Time: 48

{
  "result": {
    "id": "a6787f81343f4f1792d77c39bd972401",
    "protocol": "tcp/80",
    "dns": {
      "type": "ADDRESS",
      "name": "xxx"
    },
    "origin_dns": {
      "name": "xxx"
    },
    "origin_port": 2080,
    "ip_firewall": false,
    "proxy_protocol": "v2",
    "tls": "off",
    "traffic_type": "direct",
    "edge_ips": {
      "type": "static",
      "ips": [
        "xxx",
"xxx"
      ]
    },
    "created_on": "2021-10-21T16:44:23.977388Z",
    "modified_on": "2021-10-21T16:44:23.977388Z"
  },
  "success": true,
  "errors": [],
  "messages": []
}

-----------------------------------------------------: timestamp=2022-05-10T16:32:50.707Z
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: panic: runtime error: invalid memory address or nil pointer dereference
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb5e3d9]
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: 
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: goroutine 356 [running]:
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareSpectrumApplicationRead(0xc0005bb180, {0xd5e100, 0xc0001e4840})
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0:        github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_spectrum_application.go:116 +0xfb9
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xe9b8e0, {0xe9b8e0, 0xc0002136e0}, 0xd, {0xd5e100, 0xc0001e4840})
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0:        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/resource.go:347 +0x178
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0004ed0a0, {0xe9b8e0, 0xc0002136e0}, 0xc000845380, {0xd5e100, 0xc0001e4840})
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0:        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/resource.go:650 +0x47b
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00012f1b8, {0xe9b838, 0xc00038d100}, 0xc00038d180)
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0:        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/grpc_provider.go:613 +0x534
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00017cb40, {0xe9b8e0, 0xc000213080}, 0xc000795260)
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0:        github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/tf5server/server.go:746 +0x48c
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xd314e0, 0xc00017cb40}, {0xe9b8e0, 0xc000213080}, 0xc000795200, 0x0)
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0:        github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001468c0, {0xeaa328, 0xc0003ac4e0}, 0xc000937e60, 0xc0004f4de0, 0x13fb370, 0x0)
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0:        google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: google.golang.org/grpc.(*Server).handleStream(0xc0001468c0, {0xeaa328, 0xc0003ac4e0}, 0xc000937e60, 0x0)
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0:        google.golang.org/grpc@v1.45.0/server.go:1619 +0xa2a
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: google.golang.org/grpc.(*Server).serveStreams.func1.2()
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0:        google.golang.org/grpc@v1.45.0/server.go:921 +0x98
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0: created by google.golang.org/grpc.(*Server).serveStreams.func1
2022-05-10T16:32:50.710Z [DEBUG] provider.terraform-provider-cloudflare_v3.11.0:        google.golang.org/grpc@v1.45.0/server.go:919 +0x294
2022-05-10T16:32:50.714Z [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/3.11.0/linux_amd64/terraform-provider-cloudflare_v3.11.0 pid=297 error="exit status 2"
2022-05-10T16:32:50.714Z [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2022-05-10T16:32:50.714Z [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2022-05-10T16:32:50.714Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-05-10T16:32:50.714Z [ERROR] vertex "module.nessy.cloudflare_spectrum_application.endpoint[\"ep1:80\"]" error: Plugin did not respond
2022-05-10T16:32:50.714Z [ERROR] vertex "module.nessy.cloudflare_spectrum_application.endpoint[\"ep1:443\"]" error: Plugin did not respond
2022-05-10T16:32:50.714Z [ERROR] vertex "module.nessy.cloudflare_spectrum_application.endpoint" error: Plugin did not respond
2022-05-10T16:32:50.714Z [ERROR] vertex "module.nessy.cloudflare_spectrum_application.endpoint" error: Plugin did not respond
2022-05-10T16:32:50.714Z [ERROR] vertex "module.nessy.cloudflare_spectrum_application.endpoint (expand)" error: Plugin did not respond
2022-05-10T16:32:50.714Z [ERROR] vertex "module.nessy.cloudflare_spectrum_application.endpoint (expand)" error: Plugin did not respond
2022-05-10T16:32:50.717Z [INFO]  backend/local: plan operation completed
╷
│ Error: Plugin did not respond
│ 
│   with module.nessy.cloudflare_spectrum_application.endpoint["ep1:80"],
│   on ../../modules/proxy/main.tf line 105, in resource "cloudflare_spectrum_application" "endpoint":
│  105: resource "cloudflare_spectrum_application" "endpoint" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.nessy.cloudflare_spectrum_application.endpoint["ep1:443"],
│   on ../../modules/proxy/main.tf line 105, in resource "cloudflare_spectrum_application" "endpoint":
│  105: resource "cloudflare_spectrum_application" "endpoint" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-cloudflare_v3.11.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb5e3d9]

goroutine 356 [running]:
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareSpectrumApplicationRead(0xc0005bb180, {0xd5e100, 0xc0001e4840})
        github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_spectrum_application.go:116 +0xfb9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xe9b8e0, {0xe9b8e0, 0xc0002136e0}, 0xd, {0xd5e100, 0xc0001e4840})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/resource.go:347 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0004ed0a0, {0xe9b8e0, 0xc0002136e0}, 0xc000845380, {0xd5e100, 0xc0001e4840})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/resource.go:650 +0x47b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00012f1b8, {0xe9b838, 0xc00038d100}, 0xc00038d180)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/grpc_provider.go:613 +0x534
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00017cb40, {0xe9b8e0, 0xc000213080}, 0xc000795260)
        github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/tf5server/server.go:746 +0x48c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xd314e0, 0xc00017cb40}, {0xe9b8e0, 0xc000213080}, 0xc000795200, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001468c0, {0xeaa328, 0xc0003ac4e0}, 0xc000937e60, 0xc0004f4de0, 0x13fb370, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0001468c0, {0xeaa328, 0xc0003ac4e0}, 0xc000937e60, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1619 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.45.0/server.go:919 +0x294

Error: The terraform-provider-cloudflare_v3.11.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Panic output

_No response_

### Expected output

module.nessy.cloudflare_spectrum_application.endpoint["ep1:80"]: Refreshing state... [id=a6787f81343f4f1792d77c39bd972401]
module.nessy.cloudflare_spectrum_application.endpoint["ep1:443"]: Refreshing state... 

### Actual output

module.nessy.cloudflare_spectrum_application.endpoint["ep1:80"]: Refreshing state... [id=a6787f81343f4f1792d77c39bd972401]
module.nessy.cloudflare_spectrum_application.endpoint["ep1:443"]: Refreshing state... [id=f9b89defcc8c4b64afbfc5242ad04b61]
╷
│ Error: Request cancelled
│ 
│   with module.nessy.cloudflare_spectrum_application.endpoint["ep1:80"],
│   on ../../modules/proxy/main.tf line 105, in resource "cloudflare_spectrum_application" "endpoint":
│  105: resource "cloudflare_spectrum_application" "endpoint" {
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵
╷
│ Error: Request cancelled
│ 
│   with module.nessy.cloudflare_spectrum_application.endpoint["ep1:443"],
│   on ../../modules/proxy/main.tf line 105, in resource "cloudflare_spectrum_application" "endpoint":
│  105: resource "cloudflare_spectrum_application" "endpoint" {
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵

### Steps to reproduce

Use Cloudflare provider version 3.11.0 or newer and configure cloudflare_spectrum_application with [BYOIP](https://developers.cloudflare.com/spectrum/getting-started/byoip/)

### Additional factoids

This error appeared after upgrading Cloudflare provider to version 3.11.0 or newer

### References

_No response_

--- a/.changelog/2032.txt
+++ b/.changelog/2032.txt
@@ -1,3 +1,3 @@
 ```release-note:bug
-resource/cloudflare_spectrum_application: Keep order of edge IPs to be same 
+resource/cloudflare_spectrum_application: ignore ordering of `edge_ips`
 ```

--- a/.changelog/2032.txt
+++ b/.changelog/2032.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/cloudflare_spectrum_application: Keep order of edge IPs to be same 
+```

--- a/internal/provider/resource_cloudflare_spectrum_application.go
+++ b/internal/provider/resource_cloudflare_spectrum_application.go
@@ -288,7 +288,7 @@ func applicationFromResource(d *schema.ResourceData) cloudflare.SpectrumApplicat
 			Type: cloudflare.SpectrumEdgeTypeStatic,
 		}
 		// connectivity = cloudflare.SpectrumConnectivityStatic
-		for _, value := range edgeIPs.([]interface{}) {
+		for _, value := range edgeIPs.(*schema.Set).List() {
 			application.EdgeIPs.IPs = append(application.EdgeIPs.IPs, net.ParseIP(value.(string)))
 		}
 	}

--- a/internal/provider/resource_cloudflare_spectrum_application_test.go
+++ b/internal/provider/resource_cloudflare_spectrum_application_test.go
@@ -250,6 +250,42 @@ func TestAccCloudflareSpectrumApplication_EdgeIPsWithoutConnectivity(t *testing.
 	})
 }
 
+func TestAccCloudflareSpectrumApplication_EdgeIPsMultiple(t *testing.T) {
+	var spectrumApp cloudflare.SpectrumApplication
+	domain := os.Getenv("CLOUDFLARE_DOMAIN")
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	rnd := generateRandomResourceName()
+	name := "cloudflare_spectrum_application." + rnd
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareSpectrumApplicationConfigMultipleEdgeIPs(zoneID, domain, rnd, `"172.65.64.13", "172.65.64.14"`),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareSpectrumApplicationExists(name, &spectrumApp),
+					testAccCheckCloudflareSpectrumApplicationIDIsValid(name),
+					resource.TestCheckResourceAttr(name, "edge_ips.#", "2"),
+					resource.TestCheckTypeSetElemAttr(name, "edge_ips.*", "172.65.64.13"),
+					resource.TestCheckTypeSetElemAttr(name, "edge_ips.*", "172.65.64.14"),
+				),
+			},
+			{
+				Config: testAccCheckCloudflareSpectrumApplicationConfigMultipleEdgeIPs(zoneID, domain, rnd, `"172.65.64.14", "172.65.64.13"`),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareSpectrumApplicationExists(name, &spectrumApp),
+					testAccCheckCloudflareSpectrumApplicationIDIsValid(name),
+					resource.TestCheckResourceAttr(name, "edge_ips.#", "2"),
+					resource.TestCheckTypeSetElemAttr(name, "edge_ips.*", "172.65.64.13"),
+					resource.TestCheckTypeSetElemAttr(name, "edge_ips.*", "172.65.64.14"),
+				),
+			},
+		},
+	})
+}
+
+
 func testAccCheckCloudflareSpectrumApplicationExists(n string, spectrumApp *cloudflare.SpectrumApplication) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		rs, ok := s.RootModule().Resources[n]
@@ -431,4 +467,21 @@ resource "cloudflare_spectrum_application" "%[3]s" {
   origin_port   = 22
   edge_ips = ["172.65.64.13"]
 }`, zoneID, zoneName, ID)
+}
+
+func testAccCheckCloudflareSpectrumApplicationConfigMultipleEdgeIPs(zoneID, zoneName, ID, IPs string) string {
+	return fmt.Sprintf(`
+resource "cloudflare_spectrum_application" "%[3]s" {
+  zone_id  = "%[1]s"
+  protocol = "tcp/22"
+
+  dns {
+    type = "ADDRESS"
+    name = "%[3]s.%[2]s"
+  }
+
+  origin_direct = ["tcp://128.66.0.4:23"]
+  origin_port   = 22
+  edge_ips = [%[4]s]
+}`, zoneID, zoneName, ID, IPs)
 }

--- a/internal/provider/resource_cloudflare_spectrum_application_test.go
+++ b/internal/provider/resource_cloudflare_spectrum_application_test.go
@@ -262,29 +262,28 @@ func TestAccCloudflareSpectrumApplication_EdgeIPsMultiple(t *testing.T) {
 		ProviderFactories: providerFactories,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccCheckCloudflareSpectrumApplicationConfigMultipleEdgeIPs(zoneID, domain, rnd, `"172.65.64.13", "172.65.64.14"`),
+				Config: testAccCheckCloudflareSpectrumApplicationConfigMultipleEdgeIPs(zoneID, domain, rnd, `"172.65.64.13", "172.65.64.49"`),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckCloudflareSpectrumApplicationExists(name, &spectrumApp),
 					testAccCheckCloudflareSpectrumApplicationIDIsValid(name),
 					resource.TestCheckResourceAttr(name, "edge_ips.#", "2"),
 					resource.TestCheckTypeSetElemAttr(name, "edge_ips.*", "172.65.64.13"),
-					resource.TestCheckTypeSetElemAttr(name, "edge_ips.*", "172.65.64.14"),
+					resource.TestCheckTypeSetElemAttr(name, "edge_ips.*", "172.65.64.49"),
 				),
 			},
 			{
-				Config: testAccCheckCloudflareSpectrumApplicationConfigMultipleEdgeIPs(zoneID, domain, rnd, `"172.65.64.14", "172.65.64.13"`),
+				Config: testAccCheckCloudflareSpectrumApplicationConfigMultipleEdgeIPs(zoneID, domain, rnd, `"172.65.64.49", "172.65.64.13"`),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckCloudflareSpectrumApplicationExists(name, &spectrumApp),
 					testAccCheckCloudflareSpectrumApplicationIDIsValid(name),
 					resource.TestCheckResourceAttr(name, "edge_ips.#", "2"),
 					resource.TestCheckTypeSetElemAttr(name, "edge_ips.*", "172.65.64.13"),
-					resource.TestCheckTypeSetElemAttr(name, "edge_ips.*", "172.65.64.14"),
+					resource.TestCheckTypeSetElemAttr(name, "edge_ips.*", "172.65.64.49"),
 				),
 			},
 		},
 	})
 }
-
 
 func testAccCheckCloudflareSpectrumApplicationExists(n string, spectrumApp *cloudflare.SpectrumApplication) resource.TestCheckFunc {
 	return func(s *terraform.State) error {

--- a/internal/provider/schema_cloudflare_spectrum_application.go
+++ b/internal/provider/schema_cloudflare_spectrum_application.go
@@ -118,9 +118,9 @@ func resourceCloudflareSpectrumApplicationSchema() map[string]*schema.Schema {
 		},
 
 		"edge_ips": {
-			Type:     schema.TypeList,
+			Type:     schema.TypeSet,
 			Optional: true,
-			Elem:     &schema.Schema{Type: schema.TypeSet},
+			Elem:     &schema.Schema{Type: schema.TypeString},
 		},
 
 		"edge_ip_connectivity": {

--- a/internal/provider/schema_cloudflare_spectrum_application.go
+++ b/internal/provider/schema_cloudflare_spectrum_application.go
@@ -120,7 +120,7 @@ func resourceCloudflareSpectrumApplicationSchema() map[string]*schema.Schema {
 		"edge_ips": {
 			Type:     schema.TypeList,
 			Optional: true,
-			Elem:     &schema.Schema{Type: schema.TypeString},
+			Elem:     &schema.Schema{Type: schema.TypeSet},
 		},
 
 		"edge_ip_connectivity": {

